BuddySave/Core/ClientSession.cs

```diff
@@ -0,0 +1,27 @@
+﻿using BuddySave.Core.Models;
+using BuddySave.System;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics;
+
+namespace BuddySave.Core;
+
+public class ClientSession(IProcessProvider processProvider, ILogger<ClientSession> logger) : IClientSession
+{
+	public void StartClient(Session session, ClientParameters clientParameters)
+	{
+		if (string.IsNullOrEmpty(clientParameters.Path))
+		{
+			throw new ArgumentException("No client path provided. Cannot start a client session.");
+		}
+
+		var arguments = clientParameters.Arguments.Replace("{{Ip}}", session.Ip).Replace("{{Port}}", session.Port);
```

Comment:
Since we are validating path, IMO at least same thing should be done if not more for `Arguments` :)